libbpf-cargo/src/btf/btf.rs

```diff
@@ -22,9 +22,12 @@ pub struct Btf<'a> {
 impl<'a> Btf<'a> {
     pub fn new(name: &str, object_file: &[u8]) -> Result<Option<Self>> {
         let cname = CString::new(name)?;
-        let mut obj_opts = libbpf_sys::bpf_object_open_opts::default();
-        obj_opts.sz = std::mem::size_of::<libbpf_sys::bpf_object_open_opts>() as libbpf_sys::size_t;
-        obj_opts.object_name = cname.as_ptr();
+        let obj_opts = libbpf_sys::bpf_object_open_opts {
+            sz: std::mem::size_of::<libbpf_sys::bpf_object_open_opts>() as libbpf_sys::size_t,
+            object_name: cname.as_ptr(),
+            ..Default::default()
```

Comment:
does this ensure that all the bytes are zeroed out, even padding?

Reply:
I don't think so.

Example: https://godbolt.org/z/r75T13Pcd

Rust generates code to call `::default()` which is implemented by https://docs.rs/libbpf-sys/0.3.0-1/src/libbpf_sys/bindings.rs.html#2987-2991 and the implementation zeros out the struct using https://doc.rust-lang.org/stable/std/mem/fn.zeroed.html . From reading the docs there and looking at https://github.com/rust-lang/rust/issues/70230 , I don't believe bindgen is generating the right code here. It should probably generate a memset.

Also note that this PR doesn't change the old behavior. Functionally it should be the same as before.

Reply:
There's definitely padding: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=b7342c8fe84d6bb1e01333e4e07f7a79 (48 vs 38). I think this should be a bindgen bug

Reply:
@danobi I think I'm following along, though I'm still p new to systems programming. 

It sounds like the padding is bad because we want libbpf-rs to be as conservative as possible when it comes to memory, yeah? Also, It seems like rust-lang/rust#70230 is going to take some time to resolve, and when it does it'll be in the nightly release at first.

I can revert these changes around calling `default()` and add a comment so that `clippy` ignores em. That work?

Reply:
I think your original changes were good. I think it's a bug with bindgen (a tool used to generate rust bindings to c/c++ libraries). I'm working on fixing the bindgen bug in https://github.com/rust-lang/rust-bindgen/pull/2051

Reply:
reverted the revert!